packages/embark/src/lib/core/processes/processManager.js

```diff
@@ -21,6 +21,10 @@ class ProcessManager {
 
     this._registerApiCalls();
     this._registerEvents();
+    this.plugin.registerActionForEvent("deploy:beforeAll", (cb) => {
+      this._registerCommands();
```

Comment:
I'm not sure, but this action may be called before each deploy (in the case of modifying contracts. Just check, because registering multiple times could be a problem?

Reply:
Yes, that's very fair point indeed. I updated it to listen for the event instead.